### PR TITLE
Option to use USGS geometry

### PR DESCRIPTION
**Pull request by pjhartzell**
**Related Issue(s):**
None

**Description:**
Added ability to pull geometry from an existing USGS STAC Item in lieu of the current method that computes the geometry from the `ANG.txt` metadata file. First looks for the USGS STAC in the same location as the XML MTL metadata file. If not there, checks the USGS STAC API. If not found in either location, defaults back to using the `ANG.txt` data for computing the geometry.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x]  Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).
- [ ] Examples are added to `examples/`

### CODE REVIEW

File: tests/test_geometry.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import unittest
+
+from stactools.landsat.stac import create_stac_item
+from tests.data import TEST_GEOMETRY_PATHS
+
+
+class GeometryTest(unittest.TestCase):
+
+    def testStoredStac(self):
```
Comment (gadomski) [inline tests/test_geometry.py]:
```suggestion
    def test_stored_stac(self):
```

File: tests/test_geometry.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                             [70.70487567768816, -80.24577066106241]]]
+        }
+        expected_coords = expected_geometry["coordinates"][0]
+        item = create_stac_item(mtl_xml_href, use_usgs_geometry=True)
+        item_coords = item.geometry["coordinates"][0]
+        for e, i in zip(expected_coords, item_coords):
+            self.assertAlmostEqual(e[0], i[0], 7)
+            self.assertAlmostEqual(e[1], i[1], 7)
+
+    def testApiStac(self):
```
Comment (gadomski) [inline tests/test_geometry.py]:
```suggestion
    def test_api_stac(self):
```

File: tests/test_geometry.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                             [-124.27364628436257, 48.508467268961375]]]
+        }
+        expected_coords = expected_geometry["coordinates"][0]
+        item = create_stac_item(mtl_xml_href, use_usgs_geometry=True)
+        item_coords = item.geometry["coordinates"][0]
+        for e, i in zip(expected_coords, item_coords):
+            self.assertAlmostEqual(e[0], i[0], 7)
+            self.assertAlmostEqual(e[1], i[1], 7)
+
+    def testAng(self):
```
Comment (gadomski) [inline tests/test_geometry.py]:
```suggestion
    def test_ang(self):
```

**Review by gadomski — CHANGES_REQUESTED** [review]
Since we're using network resources in unit tests, we should probably use [vcrpy](https://vcrpy.readthedocs.io/en/latest/) to record the responses and allow unit tests to run in non-network environments. HMU if you want pointers on setting up vcrpy.

**Comment by pjhartzell** [comment]
> Since we're using network resources in unit tests, we should probably use [vcrpy](https://vcrpy.readthedocs.io/en/latest/) to record the responses and allow unit tests to run in non-network environments. HMU if you want pointers on setting up vcrpy.

Sweet, yeah, I like this. Implemented.

**Review by gadomski — APPROVED** [review]

**Pull request merged**